server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java

```diff
@@ -631,10 +631,12 @@ public void testReplicaOnNewestVersionIsPromoted() {
         assertThat(clusterState.getRoutingNodes().shardsWithState(STARTED).size(), equalTo(2));
         assertThat(clusterState.getRoutingNodes().shardsWithState(UNASSIGNED).size(), equalTo(2));
 
+        Version versionNode3 = VersionUtils.randomVersionBetween(random(),Version.CURRENT.minimumIndexCompatibilityVersion(), null);
+        Version versionNode4 = VersionUtils.randomVersionBetween(random(),Version.CURRENT.minimumIndexCompatibilityVersion(), null);
         clusterState = ClusterState.builder(clusterState).nodes(
                 DiscoveryNodes.builder(clusterState.nodes())
-                .add(newNode("node3-6.x", VersionUtils.randomVersionBetween(random(), Version.V_6_0_0_alpha1, null)))
-                .add(newNode("node4-6.x", VersionUtils.randomVersionBetween(random(), Version.V_6_0_0_alpha1, null))))
+                .add(newNode("node3-7.x" + versionNode3, versionNode3))
```

Comment:
I would change the "7.x" to "-old" or something like that, since this will automatically bump to new versions as current is updated.